src/multiserialviewer/serial_data/serialDataProcessor.py

```diff
@@ -1,52 +1,40 @@
-import time
-import typing
-from threading import Thread, Event
-from queue import Queue, Empty
-from PySide6.QtCore import Signal, QObject
+from PySide6.QtCore import Signal, Slot, QObject, QThread, QByteArray, Qt
+
+
+class Processor(QObject):
+    asciiData = Signal(str)
+
+    Slot(QByteArray)
+    def handleRawData(self, rawData: QByteArray):
+        data : bytearray = bytearray(rawData) 
+
+        if len(data) > 0:
+            self.asciiData.emit(data.decode(encoding='ascii', errors='ignore'))
 
 
 class SerialDataProcessor(QObject):
```

Comment:
I first struggled with this extra layer here, but I think the reason for this is: you don't want to have the thread handling inside the serialViewerController. Correct?
In my opinion the controller is made for this and should decide which instances are needed and should run in an extra thread. Then you don't need the private signals/slots any more. Maybe I can explain it better on the phone...

Reply:
Yes when I made this I struggled with the same... For me as reader it was more clear to offload the thread in a separate class. May I have to ask myself again if this was a good decision... I will have a look into it.

Reply:
I changed it now. But now it isn't that clear anymore which method is executed in the threads context. In fact every @Slot decorated method...

Reply:
yes, and no other non-slot method should access member data